xwayland: use _NET_WM_NAME instead of WM_NAME

puckipedia opened the pull request:
I'm a bit uncertain about this, as it'd technically break titles for any programs that don't set `_NET_WM_NAME` and only `WM_NAME`. But I'm not well-versed enough in OCaml and `Lwt` in particular to make this code flow smoothly if I have two `Lwt.t` to work with :p

talex5 commented:
(this fixes the problem that `WM_NAME` doesn't use UTF-8. e.g. try `gvim Cómo`)

You should be able to do something like this to get both in parallel:
```ocaml
  let* wm_title =
    let* atom = intern t "WM_NAME" in
    X11.Property.get_string x11 window atom
  and* net_wm_title = 
    let* atom = intern t "_NET_WM_NAME"
    in X11.Property.get_string x11 window atom
  ...
  in
  let title = 
    match wm_title, net_wm_title with ...
  in
```

puckipedia commented:
Rebased this fix, and added support for both `WM_NAME` and `_NET_WM_NAME` when creating the window; when the title is updated, however, it uses the last property update of either `WM_NAME` or `_NET_WM_NAME`  for now.

talex5 commented on xwayland.ml:
```diff
@@ -951,8 +958,9 @@ let listen_x11 ~selection t =
-          if atom = wm_name then (
+          let wm_name = intern t "WM_NAME"
+          and net_wm_name = intern t "_NET_WM_NAME" in
+          if atom = wm_name || atom = net_wm_name then (
             match Hashtbl.find_opt t.paired window with
             | Some { xdg_role = `Toplevel toplevel; _ } ->
               let title = X11.Property.get_string x11 window wm_name |> Option.value ~default:"<untitled>" in
```
This isn't quite right - it always uses `wm_name`. It should only use that if the NET property isn't being used.

puckipedia replied:
yes, that's the comment i made earlier :p this would require tracking if `_NET_WM_NAME` is set, which i haven't yet done.

talex5 replied:
It doesn't work at all, because it uses wm_name even if the property being set is net_wm_name. I've pushed a fix for that to your branch. If you're happy with that, it's ready to merge!

puckipedia replied:
Yeah, that looks good to me! Probably better than I would've written myself :p
